homework/schedule/schedule.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <chrono>
+#include <functional>
+#include <thread>
+
+template<typename T, typename ...Args>
+void schedule(T&& fun, std::chrono::duration<double> time, Args&&... args){
+    std::this_thread::sleep_for(time);
+    std::invoke(fun, std::forward<Args>(args)...);
```

Comment:
what are the advantages of using here std::invoke and std::forward?

Reply:
The std:: invoke allows you to call methods of the class. Example:
    schedule(&Foo::print, 1s, foo, 15);
The std::forward function  allows you to pass temporary values to function without copy or assigment.
The std::forward function acts as a conditional cast to T&& when the deduced template parameter is a non-reference type.
This makes the schedule function more universal to use.